public/js/index.js

```diff
@@ -524,6 +524,62 @@ Index.loadContextMenuCategories = (catList, id) => Server.callAPI(`/api/archives
     },
 );
 
+/**
+ * Build rating options for contextMenu and select the one for the current ID.
+ * @param {*} id The ID of the archive to check
+ * @returns Ratings
+ */
+Index.loadContextMenuRatings = (id) => Server.callAPI(`/api/archives/${id}/metadata`, "GET", null, `Error finding metadata for ${id}!`,
+    (data) => {
+        const items = {};
+        const ratings = [{
+            id: 1,
```

Comment:
The `id` parameters are unnecessary here, but you'll need to add `type:` if you want the click events to work.  
`type:"checkbox"` works, and you could precheck the checkbox that matches the current rating tag if there's one.

Reply:
Fixed on new commit.